contracts/defi/OperatorProxy.sol

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.2;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "./UniswapOperator.sol";
+import "./UbeswapOperator.sol";
+import "./SushiOperator.sol";
+
+contract OperatorProxy is Ownable {
+  using SafeMath for uint256;
+  using SafeERC20 for IERC20;
+
+  SushiOperator private sushiOperator;
+  UbeswapOperator private ubeswapOperator;
+
+  bool private isUbeswapInitialized = false;
+  bool private isSushiswapInitialized = false;
+
+  function setSushiOperator(
+    address _router,
+    address _factory,
+    address _miniChef,
+    address _rewarder
+  ) external onlyOwner {
+    sushiOperator = new SushiOperator(_router, _factory, _miniChef, _rewarder);
+    isSushiswapInitialized = true;
+  }
+
+  function setUbeswapOperator(
+    address _router,
+    address _factory,
+    address _rewarder
+  ) external onlyOwner {
+    ubeswapOperator = new UbeswapOperator(_router, _factory, _rewarder);
+    isUbeswapInitialized = true;
+  }
+
+  /**
+   *Sushiswap implementation
+   */
+  function swapAndZapInto(
+    address from,
+    address to,
+    uint256 fromAmount,
+    uint256 minAmountOut,
+    uint256 percentMin,
+    uint256 pid
+  ) external {
+    require(isSushiswapInitialized, "operator not initialized");
+    IERC20(from).safeTransferFrom(msg.sender, address(this), fromAmount);
+    IERC20(from).approve(address(sushiOperator), fromAmount);
```

Comment:
Este `approve` creo que debería ser un `safeApprove`. El nombre `safe` es un poco confuso en mi opinión pero lo que hace es asegurarse que el approve (u otras operaciones) no falló. 

Sino igual no termina pasando nada porque si el approve falla el zap in va a fallar después pero es más prolijo y fácil para debuguear que falle en el origen del problema :)